### ISSUE TITLE
vim9script: Support typed :let at script scope

### ISSUE THREAD

**Issue by puremourning** [vim/vim#5671]
**Is your feature request related something that is currently hard to do? Please describe.**

With vim9script's new scoping and typing rules, I would like to declare a script-scope variable with a type, such as `channel`

**Describe the solution you'd like**

When the script starts with `vim9script`, recognise (or provide alternative for) `:let ch:channel` or `let <var>:<type>[ = <value>]`. 

Currently a `let` at script scope is assumed to be a vim8-style `let` and does not accept a type.

**Describe alternatives you've considered**

I guess the workaround is to do:

```
def SetUp(): void
  let s:ch:channel
enddef

SetUp()
```

**Additional context**

To avoid the X/Y problem: I'd like to have a script-scope non-exported `channel` variable. Should I be using the `s:` dict for this ?

**Comment by puremourning** [comment]
Example:

```
vim9script

let ch:channel
```

Sourcing that gives: 

```
Error detected while processing /Users/ben/Development/vim/test.vim:
line    3:
E121: Undefined variable: ch
E488: Trailing characters
```

**Comment by puremourning** [comment]
This now works.

**Comment by lacygoill** [comment]
Can confirm that the issue was fixed by [8.2.0972](https://github.com/vim/vim/releases/tag/v8.2.0972).  As a result, this todo item is no longer relevant:

https://github.com/vim/vim/blob/ecdd14a427108aa6f133c00655788f6ae80ad463/runtime/doc/todo.txt#L80-L83

At least the first 2 lines.

---

`:const` still raises an error:

    vim -Nu NONE -S <(cat <<'EOF'
        vim9script
        const ch: channel
    EOF
    )

    E1021: const requires a value

But I don't think it would make sense to support locking a variable without assigning it a value first, so it's probably not a bug.

**Comment by brammool** [comment]
I'll update the todo list.
A "const" indeed requires a value, otherwise it can't be constant.
